PreFreeSurfer/PreFreeSurferPipeline.sh

```diff
@@ -355,6 +366,72 @@ RUN=`opts_GetOpt1 "--printcom" $@`
 UseJacobian="$(echo ${UseJacobian} | tr '[:upper:]' '[:lower:]')"
 UseJacobian=`opts_DefaultOpt $UseJacobian "true"`
 
+
+# ------------------------------------------------------------------------------
+#  Check MMP Version
+# ------------------------------------------------------------------------------
+
+Compliance="hcp"
+MPPVersion=`opts_GetOpt1 "--mppversion" $@`
+MPPVersion=`opts_DefaultOpt $MPPVersion "hcp"`
+
+if [ "${MPPVersion}" = "legacy" ] ; then
+  log_Msg "Legacy Minimal Preprocessing Pipelines: PreFreeSurferPipeline v.XX"
+  log_Msg "NOTICE: You are using MPP version that enables processing of images that do not"
+  log_Msg "        conform to the HCP specification as described in Glasser et al. (2013)!"
+  log_Msg "        Be aware that if the HCP requirements are not met, the level of data quality"
+  log_Msg "        can not be guaranteed and the Glasser et al. (2013) paper should not be used"
+  log_Msg "        in support of this workflow. A mnauscript with comprehensive evaluation for"
```

Comment:
Typo "mnauscript"

Reply:
This would be the changed version that implements all the @mharms comments:

``` bash
Compliance="standard"
MPPVersion=`opts_GetOpt1 "--mpp-mode" $@`
MPPVersion=`opts_DefaultOpt $MPPVersion "standard"`

if [ "${MPPVersion}" = "extended" ] ; then
  log_Msg "Extended Minimal Preprocessing Pipelines: PreFreeSurferPipeline"
  log_Msg "NOTICE: You are using MPP version that enables processing of images that do not"
  log_Msg "        conform to the HCP specification as described in Glasser et al. (2013)!"
  log_Msg "        Be aware that if the HCP requirements are not met, the level of data quality"
  log_Msg "        can not be guaranteed and the Glasser et al. (2013) paper should not be used"
  log_Msg "        in support of this workflow. A manuscript with comprehensive evaluation for"
  log_Msg "        the Extended MPP workflow is in active preparation and should be appropriately"
  log_Msg "        cited when published."
  log_Msg "        "
  log_Msg "Checking available data and processing options"
else
  log_Msg "HCP Minimal Preprocessing Pipelines: PreFreeSurferPipeline"
  log_Msg "Checking data compliance with HCP MPP"
fi

# -- T2w image

if [ "${T2wInputImages}" = "NONE" ] ; then
  log_Msg "-> T2w image not present (extended)"
  SkipT2wImage="YES"
  Compliance="extended"
else
  SkipT2wImage="NO"
  log_Msg "-> T2w image present (standard)"
fi

# -- Use of custom brain

if [ ! "${CustomBrain}" = "NONE" ] ; then
  log_Msg "-> Custom brain used: ${CustomBrain} (extended)"
  Compliance="extended"
else
  log_Msg "-> No custom brain used (standard)"
fi

# -- Final evaluation

if [ "${MPPVersion}" = "extended" ] ; then
  if [ "${Compliance}" = "extended" ] ; then
    log_Msg "Processing will continue using Extended MPP mode."
  else
    log_Warn "All conditions for the use of Standard HCP MPP are met. Consider using Standard HCP MPP mode instead of Extended MPP mode."
    log_Msg "Processing will continue using Exteneded MPP mode."
  fi
else
  if [ "${Compliance}" = "extended" ] ; then
    log_Err_Abort "User requested Standard HCP MPP mode. However, compliance check for use of Standard HCP MPP mode failed."
  else
    log_Msg "Conditions for the use of Standard HCP MPP mode are met."
    log_Msg "Processing will continue using Standard HCP MPP."
  fi
fi
```

Would this be ok? (The option suggested by @coalsont to move the check to a separate file as a function still remains, just the wording and reporting is in question here.)

Reply:
I agree a single separate function would be better

Reply:
Maybe we move the text into a `log_MsgMppMode` function?

Also, I like the inclusion of the parentheticals reminding which mode is in play, but rather than using the somewhat cryptic `(standard)` and `(extended)`, I think it would be clearer if we linked back to the mpp-mode parameter itself. e.g., `(--mpp-mode=standard)` or `(--mpp-mode=extended)`

Reply:
Also, typo ("Exteneded") in the "Final evaluation" code.

Reply:
I thought we were using hcp vs legacy as the terminology

Reply:
I didn't really like that terminology, per this comment:
https://github.com/Washington-University/HCPpipelines/pull/133#discussion_r329242001
so Grega came up with Standard and Extended (which I personally like).

Reply:
Let's keep it with hcp and legacy to correspond to the language we typically use when referring to these data and methods.  I don't know what standard vs extended means and don't think it will be clear to others.

Reply:
Let's get @coalsont opinion before we ask Grega to change it back.  If find "legacy" very confusing, as `--mpp-mode=legacy` implies to me a legacy (old) version of processing code.  NOT "legacy-style" data.  In the context of the log message and usage description as they currently stand, I find "standard" and "extended" to be very clear.

Reply:
legacy-style-data and hcp-style-data would be fine if necessary.  Standard and extended is not acceptable.

Reply:
Maybe `legacyStyleData` and `hcpStyleData`?

Reply:
That's fine too.

Reply:
Implemented with HCPStyleData and LegacyStyleData in a separate `mppmodecheck.shlib` script added in a separate PR <https://github.com/Washington-University/HCPpipelines/pull/137>

Reply:
Those are okay, though a bit wordy.  I had some preference for "hcp" vs "permissive", and the env variable name (or option flag) could help make it clear.

I think we should explore using an environment variable instead of an option flag, so that a user can set it once in their setup.sh instead of it needing to be added to a bunch of launcher scripts.

Reply:
@glasserm was very clear about what is acceptable so I went with LegacyStyleData and HCPStyleData as he noted else where. We need to have this as a parameter a) so that it is easy for users to set in the parameter file or at the command run depending on the study. Also, the parameter is optional, so no need to set it if the `HCPStyleData` mode is to be used.

Reply:
To some extent, the concern is that it wasn't intuitive whether "legacy" was referring to the processing methodology, or the type of data.  Naming the option `--input-mode` or `--mpp-input-mode` could solve that without as much wordiness in the values.

Since some of these issues are new (or newly prioritized), we don't necessarily have consensus on our side.

Reply:
To me, "input" is still an ambiguous term. (The arguments to any given script are "inputs", and don't necessarily relate to limitation arising from the data acquisition per se).  If you wanted to move the description more into the option name, `--data-acq-style` would be clear to me, which could be paired with 'hcp' and 'permissive' as values. (@glasserm  didn't like {standard|extended} -- not sure what he thinks of 'permissive').  Part of the problem is that some of the new features (e.g., `--custombrain` are more of a pure code extension than a limitation in the data acquisition, and in principle could very well be added as feature of the HCPpipelines proper).  All-in-all, I'm personally ok with what Grega has now.

Reply:
It isn't clear that we want the mode change to only change the data input requirements, rather than also allowing other things to deal with the issues posed by missing inputs (the bias correction option, apparently now removed, was something like this).  Thus, "input mode" does seem appropriate to me.  However, `--data-acq-style` also seems okay if all the tests it controls are closely related to the data requirements.

Reply:
I talked to Matt briefly, and he says he doesn't want to look at github again until the weekend, to not mess up its tracking of what he has read or not.  He also doesn't want to think about alternate naming of this option until then.

I think `--data-acq-style=(hcp|legacy)` is an improvement over `--mpp-mode=(HCPStyleData|LegacyStyleData)`, and is a mashup of @glasserm and @mharms preferred suggestions.  However, since we may reconsider yet again, perhaps we should move forward as if the existing naming is final, and then fix it later when we have consensus.

Reply:
Still not a fan of `legacy` as one of the option names, and in particular I really don't like any messaging along the lines of "Legacy Minimal Preprocessing Pipelines", as that clearly implies to me a legacy type of _processing_ code, rather than _acquisition_.  If we end up using "legacy" as the option name, I think the messaging needs to be along the lines of "Minimal Preprocessing Pipelines for Legacy-Style Data Acquisitions"

Reply:
I agree that "legacy minimal preprocessing pipelines" can give the wrong impression, and the wording of log messages should probably be modified.  There was a previous convesation about what to call these things in papers, which died before Matt endorsed anything that had more clarity.

However, the only time "legacy" would need to be used with my proposal is as an argument to "--data-acq-style", which makes clear from context that it is not talking about the type of processing being performed.  I don't know how Matt feels about "permissive" or "relaxed", but I don't like "standard" or "extended" because they are far too generic (to the point of being confusing), and don't have an intuitive notion of which is recommended (or may even imply that "extended" is better, which is the opposite of what we want to convey).  "hcp" vs "legacy" has a very obvious recommendation built into the wording (even more than "hcp" vs "permissive").

Reply:
So, I had another thought about this: there are only two options, in fact there will probably only ever be two options, which suggests it should really be a boolean.  So, new idea: "--legacy-acq-style=(yes|no)", default "no".

Reply:
I rather like that, although I still suggest that @sivcek not make any further changes on this particular issue until @glasserm has a chance to provide his thoughts.

As as aside, currently we have a seemingly ad-hoc mixture of some boolean flags using `TRUE|FALSE`, while others use `YES|NO`